### DISCUSSION TITLE
Anoying Calibrate Sensor window and G6 not starting up

### DISCUSSION THREAD

**Discussion by andyrozman** [NightscoutFoundation/xDrip#2149] [Q&A]
Hi !
So I changed my G6 sensor today and after 2 hours, system started flooding me with Calibrate Sensor windows. I did calibrate (I check my BG with meter and entered one value there). But now window comes back again and again, and I have red banner (Please enter two calibrations to get started!). 
If I look at Calibration data table, there are 2 calibrations there (I don't know why) and one of them is disabled. 

Now I did another calibration... I have now 4 entries there, 3 disabled one enabled.

This always worked correctly, without any prompting for calibration. I mostly do calibration after start, but it was never anoying me with window, and even when calibration is sent towards transmitter, message doesn't go away, and no BG is displayed on graph (isually there is big number in top right corner with arrow point...

My transmiter is pretty new (I think its 8M)

**Comment by Navid200** [comment]
Would you please post a screenshot of the system status page?
You can blackout the serial number if you like.

**Reply by andyrozman** [reply]
![Screenshot_20220527-170326](https://user-images.githubusercontent.com/5500133/170737135-1dac5f53-9bab-40e7-a07a-e867fdf7496b.jpg)
![Screenshot_20220527-170550](https://user-images.githubusercontent.com/5500133/170737161-65699009-826b-4639-ab82-d80b1fab6bf5.jpg)

This are both of them.

**Reply by Navid200** [reply]
Is there a reason engineering mode is enabled?

Would you please show the page at the following?
Settings &#8722;> G5/G6 Debug Settings

**Reply by andyrozman** [reply]
I have been using Xdrip for over 8 years now, so I must have had reason sometime in the past to activate it. I think the last time it was for "Hard reset of transmitter", I was using rechargable transmitter (81xx). I think it was needed for that.
![Screenshot_20220527-171453_s](https://user-images.githubusercontent.com/5500133/170739297-4c77238a-ee28-4be9-8447-09bb43f81f64.jpg)
![Screenshot_20220527-171502_s](https://user-images.githubusercontent.com/5500133/170739300-6b40182c-034d-4fb0-891f-91b37239eeb2.jpg)

**Reply by Navid200** [reply]
There is no need to enable engineering mode to perform a hard reset.
There is never a reason to keep engineering mode enabled permanently.

There is also no reason to have fallback to xDrip enabled with this transmitter.

You should also enable "Authenticate G5 before each read".

What do the logs show?

**Reply by andyrozman** [reply]
I enabled falback yesterday.

So the whole story goes like this. Yesterday my sensor that was extended (on day 14) was showing me sensor errors, so I switched to Non-native mode, then I switched back (not at once, maybe 30 minutes later), then calibrate windows started (I am not sure when excatly, if it was after I switched to non-native or when I switched back). Then I set "Fallback to xDrip" and also "Go without calibrations".

Today I started new sensors and Calibration windows doesn't go away..

Logs don't show me much... 

Switched settings: Autheticate G5 (+), Fallback to Xdrip (-).

OK, now it works... No idea what happened. Maybe Authenticate fixed it.

**Reply by Navid200** [reply]
You can never use non-native mode with this transmitter.
You can with engineering mode enabled.
If I were you, I would disable engineering mode.

**Reply by Navid200** [reply]
OK, that post can be very confusing.  Let me rewrite:

That transmitter will never work in non-native mode.  You would not have been able to attempt if you had not enabled engineering mode.

**Reply by Navid200** [reply]
You can use an old G6 in on-native mode.  But, there is no reason to use a G6 in non-native mode, which would give up factory calibration.  That's why the capability (of using a G6 in non-native mode) is behind engineering mode.

Engineering mode is meant to be used for testing only.
There are features that the developers add but are not 100% certain about.  They put it behind engineering mode so that the other developers can test and provide feedback.   But, the regular users will not be affected by the newly added feature until enough testing is done and removed from behind engineering mode.

**Reply by andyrozman** [reply]
OK. That explains much. Every time I switch to non-native mode, I get red message "Unreliable calibration" or something like that... It would be better if I got red message saying "This version of G6 transmitter doesn't support non-native mode."

**Reply by Navid200** [reply]
There is no need for that message.   If you disable native mode with a G6, it will be automatically switch back to enabled.

That does not happen in engineering mode.  Meaning if you disable native algorithm in engineering mode, it will not be automatically enabled.  
We can not add messages to help those who use engineering mode when they shouldn't.  There are so many things you can do in engineering mode that could screw everything up.  We cannot add messages for all of those almost infinite possibilities.
